src/cascadia/TerminalControl/ThrottledFunc.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "ThrottledFunc.h"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml;
+
+ThrottledFunc::ThrottledFunc(ThrottledFunc::Func func, TimeSpan delay, CoreDispatcher dispatcher) :

```

Comment:
Yes, I left `ThrottledArgFunc` in the header because it has `template` (for the type T of the argument) so we cannot put in the source file because the compiler doesn't know ahead of time which implementations (for which Ts) it will need to generate.
But the `ThrottledFunc` is not a template so I put it in the source file because it's possible.

Reply:
oh well that makes enough sense to me. Clearly, I don't know enough about templates 🥚 -> 😅. Thanks!